keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/constraints/Ec2CanaryConstraintDeployHandler.kt

```diff
@@ -46,18 +48,24 @@ class Ec2CanaryConstraintDeployHandler(
 
   override suspend fun deployCanary(
     constraint: CanaryConstraint,
+    artifact: DeliveryArtifact,
     version: String,
     deliveryConfig: DeliveryConfig,
     targetEnvironment: Environment,
     regions: Set<String>
   ): Map<String, Task> {
+    require(artifact is DebianArtifact) {
+      "Only Debian artifacts are supported for canaries on EC2"
+    }
+
     val scope = CoroutineScope(GlobalScope.coroutineContext)
     val judge = "canary:${deliveryConfig.application}:${targetEnvironment.name}:${constraint.canaryConfigId}"
 
     val images = imageService.getLatestNamedImages(
       appVersion = version.replace("~", "_").parseAppVersion(),
       account = imageResolver.defaultImageAccount,
-      regions = regions
+      regions = regions,
+      baseOs = artifact.vmOptions.baseOs
     )
```

Comment:
Another place this method was used -- deploying canaries. Makes sense to get the right base OS.